app_users/tasks.py

```diff
@@ -0,0 +1,54 @@
+import stripe
+from loguru import logger
+
+from app_users.models import PaymentProvider, TransactionReason
+from celeryapp.celeryconfig import app
+from payments.models import Subscription
+from payments.plans import PricingPlan
+from payments.webhooks import set_user_subscription
+
+
+@app.task
+def save_stripe_default_payment_method(
+    *,
+    payment_intent_id: str,
+    uid: str,
+    amount: int,
+    charged_amount: int,
+    reason: TransactionReason,
+):
+    pi = stripe.PaymentIntent.retrieve(payment_intent_id, expand=["payment_method"])
+    pm = pi.payment_method
+    if not (pm and pm.customer):
+        logger.error(
+            f"Failed to retrieve payment method for payment intent {payment_intent_id}"
+        )
+        return
+
+    # update customer's defualt payment method
+    # note: if a customer has an active subscription, the payment method attached there will be preferred
+    # see `stripe_get_default_payment_method` in payments/models.py module
+    logger.info(
+        f"Updating default payment method for customer {pm.customer} to {pm.id}"
+    )
+    stripe.Customer.modify(
+        pm.customer,
+        invoice_settings=dict(default_payment_method=pm),
+    )
+
+    # if user doesn't already have a active billing/autorecharge info, so we don't need to do anything
+    # set user's subscription to the free plan
+    if (
+        reason == TransactionReason.ADDON
+        and not Subscription.objects.filter(
+            user__uid=uid, payment_provider__isnull=False
```

Comment:
This is checking if the user has payment info stored or not. When the user unsubscribes from paypal or deletes all payment info, we clear out the payment provider

Reply:
xref 
![IMG_7702](https://github.com/user-attachments/assets/96317b94-29b1-4afc-b679-2ff0a5aece9d)

Reply:
There could be a bug when:
1. save a payment method with addon payment
2. remove the payment method from stripe
3. auto recharge is still enabled but there's no payment method that user sees in /account because payment_provider is None

i think it's not a problem though because auto recharge payments would fail (no payment method). so it's OK.

Reply:
[https://github.com/GooeyAI/gooey-server/blob/2c8ab553ac6b665ab201a20c088cf0fa29923729/payments/auto_recharge.py#L111](https://github.com/GooeyAI/gooey-server/blob/2c8ab553ac6b665ab201a20c088cf0fa29923729/payments/auto_recharge.py#L111) :)